corehq/apps/app_manager/fixtures/mobile_ucr.py

```diff
@@ -553,6 +565,11 @@ def get_report_element(
             f"Report {report_config.report_id} row count {len(rows)} exceeds max allowed row count "
             f"{settings.MAX_MOBILE_UCR_SIZE}"
         )
+    if current_row_count and len(rows) + current_row_count > settings.MAX_MOBILE_UCR_SIZE * 2:
```

Comment:
This `if` block will not execute if `current_row_count == 0`, which I think means a restore would be attempted even if `len(rows) > settings.MAX_MOBILE_UCR_SIZE * 2`.

Edit: this is addressed in ___. However, I still think it would be better and more correct to set the default of `current_row_count=0`. That would allow this initial condition to be removed.